onnxruntime/core/providers/coreml/builders/impl/base_op_builder.cc

```diff
@@ -126,6 +127,24 @@ bool BaseOpBuilder::IsInputDtypeSupport(const Node& node, size_t idx,
   return false;
 }
 
+/* static */
+bool BaseOpBuilder::CheckShapeForConvMemoryLimit(onnxruntime::VectorInt64& shape, const logging::Logger& logger) {
```

Comment:
for `shape`, we don't need to pass the vector by reference as we don't modify it. consider using `gsl::span<const int64_t>` as it allows more flexibility for the input.